### PR TITLE
Refactor existing stream error type component to stream completion type

### LINKED ISSUE #30047: Refactor existing `stream error type` component to `stream completion type`
**Description:**
$title. According to [1], in `stream<T,C>`, `C` is now considered the completion type. However, in the internal implementation, it is referred to as the error type. That must change.

[1] https://ballerina.io/spec/lang/master/#section_5.5.7

**Related Issues (optional):**
- https://github.com/ballerina-platform/ballerina-lang/pull/29955 (only fixes semantics)

### PR DESCRIPTION
**Pull request by grainier**
## Purpose
$title.

Fixes #30047

## Approach
n/a

## Samples
n/a

## Remarks
-

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by hasithaa — DISMISSED** [review]

**Comment by hasithaa** [comment]
@warunalakshitha @manuranga @rdhananjaya Can you check runtime-related changes?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/TypeCreator.java [hunk 1]
```diff
@@ -231,22 +231,25 @@ public static ObjectType createObjectType(String typeName, Module module, long f
      * Creates a {@link BStreamType} which represents the stream type.
      *
      * @param constraint the type by which this stream is constrained
+     * @param completionType the type which indicates the completion of the stream
      * @return the new stream type
      */
-    public static StreamType createStreamType(Type constraint) {
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/TypeCreator.java:236]: Should we deprecate these and add new methods instead?

File: langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/Construct.java [hunk 2]
```diff
@@ -25,21 +25,23 @@
 import io.ballerina.runtime.api.values.BTypedesc;
 
 /**
- * Native implementation of lang.internal:construct(typeDesc, iterator).
+ * Native implementation of lang.internal:construct(typeDesc, typeDesc, iterator).
  *
  * @since 1.2.0
  */
 //@BallerinaFunction(
```
Comment (MaryamZi) [inline langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/Construct.java:32]: We can remove this, right?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/StreamValue.java [hunk 3]
```diff
@@ -87,7 +90,7 @@ public Type  getIteratorNextReturnType() {
      * @param parent The link to the parent node
      */
     public String stringValue(BLink parent) {
-        return "stream <" + getType().toString() + ">";
+        return getType().toString();
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/StreamValue.java:93]: In cases where ToString is not defined, we seem to be having the basic type prefix, a space, and then some string representation. Should we continue to do the same here?

For example,

```ballerina
import ballerina/io;

public function main() {
    stream<int> st = ([1, 2]).toStream();
    io:println(typeof st);
    io:println(st);
    
    future<()> fn = start func();
    io:println(typeof fn);
    io:println(fn);
}

function func() {

}
```

currently prints

```
typedesc stream<()>
stream <stream<()>>
typedesc future
future {isDone:false}
```

Comment (grainier) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/StreamValue.java]: Revert: 6dda6a0

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/TypeCreator.java (hunk 1 above)
Comment (grainier) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/TypeCreator.java:236]: Fixed: 5a6d295

**Comment by manuranga** [comment]
Runtime aspects looks good to me.
Please add a usecase to BIR parser unit test (ballerina-lang/docs/bir-spec/src/test/resources/test-src) and change change the bir definition file (.ksy) accordingly.

**Review by pcnfernando — DISMISSED** [review]

**Comment by pcnfernando** [comment]
@grainier I think you have to bump the BIR version in https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileConstants.java

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**